plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/GlueIcebergTableOperations.java

```diff
@@ -102,7 +102,9 @@ protected String getRefreshedLocation(boolean invalidateCaches)
     protected void commitNewTable(TableMetadata metadata)
```

Comment:
Ok, I see how this is working now. I think I'd suggest something a little different. Can we have a new method `registerTable` here as well. The APIs would look something like this:

```
@Override 
protected void commitNewTable(TableMetadata metadata) 
{
    verify(metadata.metadataFileLocation() == null, "Error message...");
    doCommit(metadata, writeNewMetadata(metadata, 0));
}

@Override 
protected void registerTable(TableMetadata metadata)
{
    verify(metadata.metadataFileLocation() != null, "Error message ...");
    doCommit(metadata, metadata.metadataFileLocation());
}

private void doCommit(TableMetadata metadata, String metadataLocation)
{
    // Mostly the body of `comitNewTable` as it is today
}
```

Reply:
So for now, I took one of the approaches where we will be not going to register table through TableOperations., But we will be using TrinoCatalog to register the table. Let's see how this looks like.

CC: @alexjo2144 | @findinpath | @findepi

Reply:
I'm on the fence. Not having to call TableOperations is nice, I guess, but it duplicates the table registering code between TableOperations and the Catalog. Personally, I think I preferred having it in TableOps

Reply:
The  code duplication, I think, can be actually very much minimised.

I would argue that the current approach of having the table registration done directly from the catalog is easier to follow and to (eventually) maintain.